src/pmdas/hdb/pmdahdb.python

```diff
@@ -0,0 +1,3173 @@
+#!/usr/bin/env pmpython
+""" Performance Metric Domain Agent for SAP HANA.
+
+Performance Metric Domain Agent for SAP HANA.
+https://pcp.io/
+https://www.sap.com/products/hana.html
+
+Copyright (c) 2021 Red Hat.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later
+version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+See <http://www.gnu.org/licenses/> for copy of the GNU General Public License.
+"""
+
+# pylint: disable=too-many-arguments, too-many-positional-arguments, too-many-lines
+
+import argparse
+import configparser
+import os.path
+import sys
+from setuptools.version import LooseVersion
+from typing import Any, Dict, List, Optional
+
+from cpmapi import (
+    PM_COUNT_ONE,
+    PM_ERR_AGAIN,
+    PM_ERR_INST,
+    PM_ERR_PMID,
+    PM_INDOM_NULL,
+    PM_SEM_COUNTER,
+    PM_SEM_DISCRETE,
+    PM_SEM_INSTANT,
+    PM_SPACE_BYTE,
+    PM_SPACE_GBYTE,
+    PM_TIME_MSEC,
+    PM_TIME_SEC,
+    PM_TIME_USEC,
+    PM_TYPE_32,
+    PM_TYPE_64,
+    PM_TYPE_DOUBLE,
+    PM_TYPE_FLOAT,
+    PM_TYPE_STRING,
+    PM_TYPE_U32,
+    PM_TYPE_U64,
+)
+from hdbcli import dbapi
+from pcp.pmapi import pmContext, pmUnits
+from pcp.pmda import PMDA, pmdaIndom, pmdaInstid, pmdaMetric
+from pyhdbcli import OperationalError, ResultRow
+
+
+class HDBConnection:
+    """
+    Wrapper class for a connection to HANA.
+    """
+
+    def __init__(self, address: str, port: int, user: str, password: str):
+        # Connection parameters of the driver are documented here
+        # https://help.sap.com/viewer/f1b440ded6144a54ada97ff95dac7adf/2.10/en-US/ee592e89dcce4480a99571a4ae7a702f.html
+        self._conn = dbapi.connect(
+            address=address,
+            port=port,
+            user=user,
+            password=password,
+            databaseName=None,
+            # Controls whether the system automatically reconnects to the database instance after a command timeout or
+            # when the connection is broken. Reconnecting restores the old state.
+            reconnect=True,
+            # limits the overall time for the entire connection to try all server addresses specified.
+            # See nodeConnectTimeout for per-node timeout.
+            connectTimeout=60 * 1000,
+            # Aborts the connection attempt after the specified maximum timeout
+            nodeConnectTimeout=10 * 1000,
+            # wait for a maximum of the specified number of milliseconds for any request to complete. If the time has
+            # expired and a response has not been received, the driver returns an error.
+            communicationTimeout=2 * 1000,
+        )
```

Comment:
these timeouts seem like things that should be optionally present in the configuration file as well.